src/index.rs

```diff
@@ -386,7 +381,9 @@ impl ResourceIndex {
                             }
 
                             // the caller must have ensured that the path was updated
-                            return Err(Error::msg("Nothing to update"));
+                            return Err(ArklibError::Collision(
+                                "Nothing to update".into(),
```

Comment:
Could you also change the message to `New content has the same id.`

Reply:
I updated the message